crates/kornia-pnp/src/epnp.rs

```diff
@@ -151,15 +155,38 @@ pub fn solve_epnp(
         best_r[2][2],
     ]);
     let rvec_f32 = SO3::from_matrix(&mat).log();
-    let rvec = [rvec_f32.x, rvec_f32.y, rvec_f32.z];
+    let mut rvec = [rvec_f32.x, rvec_f32.y, rvec_f32.z];
+    let mut tvec = best_t;
+
+    let mut rmse = best_err;
+    let mut iterations = None;
+    let mut converged = Some(true);
+
+    if let Some(lm) = &params.refine_lm {
+        if let Ok((rmse_new, iters, conv)) =
+            refine_pose_lm(points_world, points_image, k, &mut rvec, &mut tvec, lm)
+        {
+            // Update rotation from refined rvec
+            let r_mat_ref = SO3::exp(glam::Vec3A::from_array(rvec)).matrix();
+            let r_ref = [
+                [r_mat_ref.x_axis.x, r_mat_ref.y_axis.x, r_mat_ref.z_axis.x],
+                [r_mat_ref.x_axis.y, r_mat_ref.y_axis.y, r_mat_ref.z_axis.y],
+                [r_mat_ref.x_axis.z, r_mat_ref.y_axis.z, r_mat_ref.z_axis.z],
+            ];
+            best_r = r_ref;
+            rmse = rmse_new;
+            iterations = Some(iters);
```

Comment:
it feels a bit weird in terms of api that iterations and converged are optional. I would upgrade the Result struct packing the this two 

```rust
#[derive(Debug, Clone)]
pub struct RefinementResult {
   pub num_iterations: usize,
   pub converged: boo,
}

#[derive(Debug, Clone)]
pub struct PnPResult {
    /// Estimated rotation matrix.
    pub rotation: [[f32; 3]; 3],
    /// Estimated translation vector.
    pub translation: [f32; 3],
    /// Rodrigues axis-angle representation of the rotation.
    pub rvec: [f32; 3],
    /// Root-mean-square reprojection error in pixels (if computed).
    pub reproj_rmse: Option<f32>,
    /// refinenement result
    pub refinement: Option<RefinementResutlt>
}
```